src/MongoDB/ServerDescription.c

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright 2021-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <php.h>
+#include <Zend/zend_interfaces.h>
+#include <ext/standard/php_var.h>
+#include <zend_smart_str.h>
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "phongo_compat.h"
+#include "php_phongo.h"
+
+zend_class_entry* php_phongo_serverdescription_ce;
+
+/* {{{ proto array MongoDB\Driver\ServerDescription::getHelloResponse()
+   Returns the most recent “hello” response */
+static PHP_METHOD(ServerDescription, getHelloResponse)
+{
+	php_phongo_serverdescription_t* intern;
+	const bson_t*                   helloResponse;
+
+	intern = Z_SERVERDESCRIPTION_OBJ_P(getThis());
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	helloResponse = mongoc_server_description_hello_response(intern->server_description);
+
+	if (helloResponse->len) {
+		php_phongo_bson_state state;
+
+		PHONGO_BSON_INIT_DEBUG_STATE(state);
+
+		if (!php_phongo_bson_to_zval_ex(bson_get_data(helloResponse), helloResponse->len, &state)) {
+			zval_ptr_dtor(&state.zchild);
+			return;
+		}
+
+		RETURN_ZVAL(&state.zchild, 0, 1);
+	} else {
+		RETURN_NULL();
+	}
+} /* }}} */
+
+/* {{{ proto string MongoDB\Driver\ServerDescription::getHost()
+   Returns the server’s hostname */
+static PHP_METHOD(ServerDescription, getHost)
+{
+	php_phongo_serverdescription_t* intern;
+
+	intern = Z_SERVERDESCRIPTION_OBJ_P(getThis());
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	RETVAL_STRING(mongoc_server_description_host(intern->server_description)->host);
+} /* }}} */
+
+/* {{{ proto integer MongoDB\Driver\ServerDescription::getLastUpdateTime()
+   Returns the server’s last update time, in microseconds */
+static PHP_METHOD(ServerDescription, getLastUpdateTime)
+{
+	php_phongo_serverdescription_t* intern;
+
+	intern = Z_SERVERDESCRIPTION_OBJ_P(getThis());
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	RETVAL_LONG((zend_long) mongoc_server_description_last_update_time(intern->server_description));
```

Comment:
Ok makes sense. I used the cast because I saw that it was used in couple other places for the round trip time: https://github.com/mongodb/mongo-php-driver/blob/80f88146f82d5d3f52be16b60742820b2b55abdb/php_phongo.c#L1209
and https://github.com/mongodb/mongo-php-driver/blob/80f88146f82d5d3f52be16b60742820b2b55abdb/src/MongoDB/Server.c#L353
Should I change it in those functions as well since the explicit cast isn't necessary?

Reply:
Looks like both of those instances date back to https://github.com/mongodb/mongo-php-driver/commit/c795d20517944518ca592d12f6c318f8dc3abafd when we had a `phongo_long` type that was conditionally defined as a `zend_long` or `long` for PHP 7 and 5, respectively (see https://github.com/mongodb/mongo-php-driver/blob/c795d20517944518ca592d12f6c318f8dc3abafd/phongo_compat.h#L163).

That was removed in https://github.com/mongodb/mongo-php-driver/commit/b7e4a8e718d294e8e68d81c8595ad3e5d12394c2 but it looks like the commit made a broad search and replace of `phongo_long` with `zend_long`, which led to this being left over. I'd suggest making a new JIRA ticket for that and marking it as related to PHPC-1442. Feel free to open a separate PR on master if you like.

Reply:
https://jira.mongodb.org/browse/PHPC-1928